File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ret = 0
        l = prices[0]
        for r in range(1,len(prices)):
            print(f"l: {l}, prices[r]: {prices[r]}")
            ret = max(ret,prices[r]-l)
            l = min(prices[r],l)
        
        return ret if ret !=float("-inf") else 0 

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    let ret = 0
    let l = prices[0]
    for (let r= 1; r<prices.length; r++){
        ret = Math.max(ret,prices[r]-l )
        l = Math.min(l,prices[r])
    }
    return ret
};